### DISCUSSION TITLE
HCS200

### DISCUSSION THREAD

**Discussion by 8um1c4** [merbanan/rtl_433#2518] [General]
The HCS200 signals as is a Keelog signals has associated to it a counter. I don't see the counter decoded in the messages. Is it possible to decode?
Thanks.

**Comment by zuckschwerdt** [comment]
We output all data from the packet. There is no additional counter?
https://github.com/merbanan/rtl_433/blob/c94ee770cfa5a1ad7a123723c901e6c63d6d27e1/src/devices/hcs200.c#L16-L20

**Comment by 8um1c4** [comment]
You can see here an example of a keelog  decoded
https://forum.flipperzero.one/uploads/default/original/2X/0/052937bb532d86e80643addaf5b275f24217a1a0.png

The counter is there to take care of syncronous between gate and control.

**Reply by zuckschwerdt** [reply]
It seems that (counter) data is only available after decoding the encrypted payload, see
https://github.com/flipperdevices/flipperzero-firmware/blob/14777a7585313d413364eb6353b047a719746ddf/lib/subghz/protocols/subghz_protocol_keeloq_common.c#L22-L33
